scripts/ai/AIDriveStrategyChopperCourse.lua

```diff
@@ -0,0 +1,782 @@
+function AIDriveStrategyChopperCourse:clearAllUnloaderInformation()
+    self:debug('All Unloader Info has been cleared')
+    self:cancelRendezvous()
+    self.unloader:reset()
+end
+
+-- This is needed for the deregisterUnloader function we don't want to cancel the rendezvous if we are not the one we currently are set to rendezvous with(logic breaks)
+function AIDriveStrategyChopperCourse:getUnloader(driver)
+    if driver == self:getCurrentUnloader() then
+        return self:getCurrentUnloader()
+    elseif driver == self:getNextUnloader() then
+        return self:getNextUnloader()
+    end
+end
+
+-- This is called any time our NextUnloader is ready to replace out current.(Within a certian distance and aligned) It checks to make sure we have a current and if we do we clear it out
+-- (The current unloader isn't always johny on the spot telling the chopper to clear itself out)
+-- Then we register as the current and reset the nextUnloader
+function AIDriveStrategyChopperCourse:updateNextUnloader()
+    if self:getCurrentUnloader() then
+        self:resetUnloader(self:getCurrentUnloader())
+    end
+    if self:getNextUnloader() then
+        self:registerUnloader(self:getNextUnloader())
+        self.unloaders.nextUnloader:reset()
+    end
+end
+
+-- Returns our current unloader. So we don't have to access the unloaders object directly
+function AIDriveStrategyChopperCourse:getCurrentUnloader()
+    return self.unloaders.currentUnloader:get()
+end
+
+-- Returns our next unloader. So we don't have to access the unloaders object directly
+function AIDriveStrategyChopperCourse:getNextUnloader()
+    return self.unloaders.nextUnloader:get()
+end
+
+-- Another saftey check to make sure we always have a current unloader
+-- This is ran every waypointpassed() call. It asks if our next unloader is behind and aligned and within 25 meters of lastwaypoint of its driveToCombine Course
+function AIDriveStrategyChopperCourse:checkNextUnloader()
+    if not self:getCurrentUnloader() and self:getNextUnloader() then
+        self:debug('checkNextUnloader: I lost my current unloder and I have one that is arriving switch them')
+        self:updateNextUnloader()
+    elseif self:getNextUnloader() and self:getNextUnloader():readyToRecive() then
+        self:debug('checkNextUnloader: Discharging to %s, and %s is ready to come along side', CpUtil.getName(self:getCurrentUnloader().vehicle), CpUtil.getName(self:getNextUnloader().vehicle))
+        --TODO This is the user method to request drive now. It resets after 3 secs which may cause and already full unloader to return if there is a reason for it go to idle state from drive to unload
+        self:getCurrentUnloader():requestDriveUnloadNow()
+        self:updateNextUnloader()
+    end
+end
+
+
+-- Coppied from parent. Altered to handle a current and next unloader
+function AIDriveStrategyChopperCourse:callUnloaderWhenNeeded()

```

Comment:
This has way too many lines common with the parent, if that really is needed, it should be split so the business logic of finding a rendezvous point is not duplicated.